src/query/plan/operator.cpp

```diff
@@ -4045,8 +4046,14 @@ class DistinctCursor : public Cursor {
 
       utils::pmr::vector<TypedValue> row(seen_rows_.get_allocator().GetMemoryResource());
       row.reserve(self_.value_symbols_.size());
-      for (const auto &symbol : self_.value_symbols_) row.emplace_back(frame[symbol]);
-      if (seen_rows_.insert(std::move(row)).second) return true;
+
+      for (const auto &symbol : self_.value_symbols_) {
+        row.emplace_back(frame[symbol]);
```

Comment:
```suggestion
        row.emplace_back(frame.at(symbol));
```

I would suggest this only because operator [] initializes new value if the key doesn't exist while at will throw an exception